src/cairoUtilFuncGen/storage/copyToStorage.ts

```diff
@@ -122,71 +123,182 @@ export class StorageToStorageGen extends StringIndexedFuncGen {
 
     return {
       name: funcName,
       code: [
-        `func ${funcName}_elem${implicits}(fromLoc: felt, toLoc: felt, index: felt) -> (retLoc: felt):`,
-        `    if index == ${type.size}:`,
-        `        return (toLoc)`,
+        `func ${funcName}_elem${implicits}(toLoc: felt, fromLoc: felt, index: felt) -> ():`,
+        `    if index == ${narrowBigIntSafe(toType.size)}:`,
+        `        return ()`,
         `    end`,
-        `    ${elementCopyFunc}(fromLoc + index, toLoc + index)`,
-        `    return ${funcName}_elem(fromLoc, toLoc, index + 1)`,
+        `    ${copyCode('toLoc', 'fromLoc')}`,
+        `    return ${funcName}_elem(toLoc + ${toElemType.width}, fromLoc + ${fromElemType.width}, index + 1)`,
         `end`,
-        `func ${funcName}${implicits}(fromLoc: felt, toLoc: felt) -> (retLoc: felt):`,
-        `    return ${funcName}_elem(fromLoc, toLoc, 0)`,
+        `func ${funcName}${implicits}(toLoc: felt, fromLoc: felt) -> (retLoc: felt):`,
+        `    ${funcName}_elem(toLoc, fromLoc, 0)`,
+        `    return (toLoc)`,
         `end`,
       ].join('\n'),
     };
   }
 
-  private createDynamicArrayCopyFunction(funcName: string, type: ArrayType): CairoFunction {
-    assert(type.size === undefined, 'Attempted to copy storage static array as dynamic array');
+  private createDynamicArrayCopyFunction(
+    funcName: string,
+    toType: ArrayType,
+    fromType: ArrayType,
+  ): CairoFunction {
+    assert(toType.size === undefined, 'Attempted to copy to storage static array as dynamic array');
+    assert(
+      fromType.size === undefined,
+      'Attempted to copy from storage static array as dynamic array',
+    );
 
     this.requireImport('starkware.cairo.common.uint256', 'Uint256');
     this.requireImport('starkware.cairo.common.uint256', 'uint256_sub');
 
-    const elementCopyFunc = this.getOrCreate(type.elementT);
-    const elementCairoType = CairoType.fromSol(
-      type.elementT,
+    const elementCopyFunc = this.getOrCreate(toType.elementT, fromType.elementT);
+    const fromElementCairoType = CairoType.fromSol(
+      fromType.elementT,
       this.ast,
       TypeConversionContext.StorageAllocation,
     );
-    const [elementMapping, lengthMapping] = this.dynArrayGen.gen(elementCairoType);
+    const toElementCairoType = CairoType.fromSol(
+      fromType.elementT,
+      this.ast,
+      TypeConversionContext.StorageAllocation,
+    );
+    const [fromElementMapping, fromLengthMapping] = this.dynArrayGen.gen(fromElementCairoType);
+    const [toElementMapping, toLengthMapping] = this.dynArrayGen.gen(fromElementCairoType);
+
+    const copyCode = createElementCopy(toElementCairoType, fromElementCairoType, elementCopyFunc);
 
     return {
       name: funcName,
       code: [
-        `func ${funcName}_elem${implicits}(fromLoc: felt, toLoc: felt, length: Uint256) -> ():`,
+        `func ${funcName}_elem${implicits}(toLoc: felt, fromLoc: felt, length: Uint256) -> ():`,
```

Comment:
Just making sure you are aware that this function is writing in the array from end to beginning since the length and index  are the same value

Reply:
Yep, no special reason other than that it was nicer to write the function like that